GeoJSON Previewer

spilth opened the pull request:
### Description

This adds a Leaflet map preview for GeoJSON files.


### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Frontend**

- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.


**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### Screenshot

<img width="871" alt="Screenshot 2025-02-24 at 8 28 50 AM" src="https://github.com/user-attachments/assets/2c37daa4-f85a-41a3-a09a-a8b52717e751" />

Commit:
feat: GeoJSON support

tmorrell commented:
This looks like a good addition, once you've finished everything up. Leaflet is BSD-2 licensed, which shouldn't be an issue.

spilth commented:
@tmorrell Should I make this PR against v2.2.2? I can't seem to get a version of Invenio RDM running against this branch due to dependency mismatches.

tmorrell commented:
No the PR should always go against master. If you're running a v12 repository the fix will need to be backported to the appropriate version branch, so you may want to do that yourself for local testing. But for testing the master branch you'll want to use the latest version of InvenioRDM...which I think should be app-rdm 13.0.0b2

spilth commented:
I managed to get it working by using `invenio-cli init rdm -c master`.

spilth commented:
@tmorrell Once I was able to test it against a V12/2.2.X version I discovered some CSS issues. Added a commit to address them and also added a screenshot to show what the previewer looks like.

tmorrell reviewed (approved):
Could you add your copyright statement to any file that has more than 15 modified lines (and not copy the CERN one to new files?) Thanks!

spilth commented:
@tmorrell Sorry for the delayed reply. I can't find other copyright statements from individual contributors, so I'm not sure what it should look like.

Are you looking for me to just add `Copyright 2025 Brian Kelly` to new files and files with more than 15 new/modified lines?

spilth commented:
@tmorrell Thank you. Done!

spilth commented:
Is there still an interest in this functionality?
